src/PermissionMatcher.php

```diff
@@ -147,6 +156,26 @@ protected function flatten($input)
     }
     
     /**
+     * Populate an array of privileges with their corresponding authz groups
+     *
+     * @param  array          $privileges            The resulting array
+     * @param  string|array   $authzGroupsPrivileges The privileges that the authzgroup has
+     * @param  array          $authzGroups
+     * @return array          $priviliges
+     */
+    protected function addAuthzGroupsToPrivileges(array $privileges, $authzGroupsPrivileges, array $authzGroups)
+    {
+        $authzGroupsPrivileges = !is_string($authzGroupsPrivileges) ? $authzGroupsPrivileges : [$authzGroupsPrivileges];
+        
+        foreach($authzGroupsPrivileges as $privilige) {
+            $privileges[$privilige] = !empty($privileges[$privilige]) ? $privileges[$privilige] : [];
+            $privileges[$privilige] = array_unique(array_merge($privileges[$privilige], $authzGroups));
```

Comment:
Rather not write intermediate values to the array. Also, mind the spelling in `$privilige`.

```php
foreach ($authzGroupsPrivileges as $privilege) {
  $current = !empty($privileges[$privilege]) ? $privileges[$privilege] : [];
  $privileges[$privilege] = array_unique(array_merge($current, $authzGroups));
}
```